app/models/endpoint.rb

```diff
@@ -3,6 +3,7 @@ class Endpoint < ApplicationRecord
 
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
+  validates :port, :numericality => {:only_integer => true, :allow_nil => true}
```

Comment:
```greater_than``` or ```greater_than_or_equal_to``` ? according to [linux kernel source]/net/ipv4/inet_connection_sock.c on inet_csk_get_port() ```Obtain a reference to a local port for the given sock, if snum is zero it means select any available local port.```